base/stream.jl

```diff
@@ -617,7 +617,7 @@ end
 function alloc_request(buffer::IOBuffer, recommended_size::UInt)
     ensureroom(buffer, Int(recommended_size))
     ptr = buffer.append ? buffer.size + 1 : buffer.ptr
-    nb = min(length(buffer.data)-buffer.offset, buffer.maxsize) + buffer.offset - ptr + 1
+    nb = min(length(buffer.data), buffer.maxsize) - ptr + 1
```

Comment:
This is now only lets writing up to `maxsize-offset` bytes, which doesn't seem intended

Reply:
It seems to be correct, yes.
1. We can't write pas `maxsize` when `offset` is zero, because if we do, then resizing or replacing the buffer (in an unrelated operation) may cause data loss, since the buffer is now smaller.
2. The call to `ensureroom` ensures there is at least `recommended_size` bytes available. If offset is not zero, the offset is zeroed (moving the content of the data).

Therefore, the pointer and length returned by `alloc_request` returns the correct maximum number of bytes that can be written, at least `recommended_size`, and only shifts the data inside the buffer if necessary to achieve `recommended_size`.

Reply:
maxsize - ptr is a negative number (equal to `-offset`) when you first created the IOBuffer with maxsize bytes available, so that can't be right

Reply:
The value `length(buffer.data)` includes offset, but buffer.maxsize does not, so these two input values to `min` do not have the same units

Reply:
Yes, for the case now using `get_offset`